src/model/attendance-board.test.ts

```diff
@@ -0,0 +1,38 @@
+import { ID } from "./id";
+
+type AttendanceBoard = {
+    readonly id: ID<AttendanceBoard>;
+    readonly subject: ID<Subject>;
+    readonly startFrom: Date;
+    readonly secondsFromStartToBeLate: number;
+    readonly secondsFromBeLateToEnd: number;
+};
+
+type AttendanceState = 
+    | "TOO_EARLY"
+    | "ATTENDED"
+    | "BE_LATE"
+    | "CLOSED"
+;
+
+function determineState(board: AttendanceBoard, toAttendAt: Date): AttendanceState {
+       
+const currentDateTime = new Date(); // 現在の日時を取得
+const Attendtime = new Date(toAttendAt.getTime() + board.secondsFromStartToBeLate * 1000);//出席と認められる時間
+const Latetime = new Date(toAttendAt.getTime() + board.secondsFromStartToBeLate * 1000 + board.secondsFromBeLateToEnd * 1000);//遅刻と認められる時間
```

Comment:
```suggestion
const currentMs = Date.now();
const beLateMs = toAttendAt.getTime() + board.secondsFromStartToBeLate * 1000;
const endMs = toAttendAt.getTime() + board.secondsFromStartToBeLate * 1000 + board.secondsFromBeLateToEnd * 1000;
```
わざわざ `Date` に変換せずとも, 最初からすべてミリ秒にして判定したほうが楽かと思います